python/rmm/_lib/memory_resource.pyx

```diff
@@ -168,6 +182,12 @@ cdef class DeviceMemoryResource:
     cdef device_memory_resource* get_mr(self):
         return self.c_obj.get()
 
+    def allocate(self, size_t nbytes):
+        return <uintptr_t>self.c_obj.get().allocate(nbytes)
```

Comment:
Why is this a `uintptr_t` instead of something like `void*`? Does it have to do with alignment? (Same question for `uintptr_t` in the deallocate call.)

Reply:
A `def` function cannot return a `void*`

Reply:
To expand Ashwin's answer, the returned value would be further cast to a Python int. But it could not happen without casting to `uintptr_t` first.

Reply:
Awesome, thank you both for the explanation!